open-api/rest-catalog-open-api.yaml

```
@@ -541,6 +541,216 @@ paths:
+        If the plan is not completed, returns a `plan-status` representing the state of the plan.
+        If the plan is completed, returns a list of `PlanTask` or `FileScanTask`.
+      
+        If an invalid `plan-id` is provided, the service will return a `404 NoSuchPlanIdError` exception indicating that the requested plan does not exist.
+      responses:
+        200:
+          $ref: '#/components/responses/GetTasksStatusResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        404:
+          description:
+            Not Found
+            - NoSuchPlanIdException, the plan-id does not exist
+            - NoSuchTableException, the table does not exist
+            - NoSuchNamespaceException, the namespace does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergErrorResponse'
+              examples:
+                PlanIdDoesNotExist:
+                  $ref: '#/components/examples/NoSuchPlanIdError'
+                TableDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        419:
+          $ref: '#/components/responses/AuthenticationTimeoutResponse'
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+    delete:
+      tags:
+        - Catalog API
+      summary: Uses a `plan-id` as input to cancel a `planTable` operation.
+      operationId: CancelPlan
+      description:
+        Uses a `plan-id` as input to cancel a `planTable` operation.
+        If successful, should return a state that says "cancelled".
```

Does this API need to return a state? I feel like a 204 status code is sufficient to indicate the cancellation was successful. And we know that for this API the only valid state change that can happen is "Started -> Cancelled"..

---

I think we can likely manage with http status codes instead of returning an explicit state to signal if the cancellation was successful or failed.

---

I updated this to 204